pkg/metrics/podlabelmetrics.go

```diff
@@ -29,6 +37,40 @@ func (kpmc KubePodLabelsCollector) Describe(ch chan<- *prometheus.Desc) {
 	}
 }
 
+func (kpmc *KubePodLabelsCollector) UpdateControllerSelectorsCache() {
```

Comment:
Do we need anything Deployment-specific here? Or does the ReplicaSet case cover Deployments adequately? Looking over `pkg/costmodel/allocation.go` the only relevant metrics I see are:
- service_selector_labels
- deployment_match_labels
- statefulSet_match_labels

Reply:
replicasets covers deployments!